src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingUseXmlReaderForXPathDocumentTests.cs

```diff
@@ -0,0 +1,366 @@
+        Finally
+        End Try
+    End Sub
+End Class",
+                GetCA3075XPathDocumentBasicResultAt(9, 24, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseXPathDocumentInFinallyBlockShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Xml.XPath;
+
+class TestClass
+{
+    private void TestMethod()
+    {
+        try { }
+        catch (Exception) { throw; }
+        finally
+        {
+            var xml = """";
+            XPathDocument doc = new XPathDocument(xml);
+        }
+    }
+}",
+                GetCA3075XPathDocumentCSharpResultAt(14, 33, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml.XPath
+
+Class TestClass
+    Private Sub TestMethod()
+        Try
+        Catch generatedExceptionName As Exception
+            Throw
+        Finally
+            Dim xml = """"
+            Dim doc As New XPathDocument(xml)
+        End Try
+    End Sub
+End Class",
+                GetCA3075XPathDocumentBasicResultAt(11, 24, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseXPathDocumentInAsyncAwaitShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Threading.Tasks;
+using System.Xml.XPath;
+
+class TestClass
+{
+    private async Task TestMethod()
+    {
+        await Task.Run(() => {
+            var xml = """";
+            XPathDocument doc = new XPathDocument(xml);
+        });
+    }
+
+    private async void TestMethod2()
+    {
+        await TestMethod();
+    }
+}",
+                GetCA3075XPathDocumentCSharpResultAt(11, 33, "Run")
+            );
+
+            VerifyBasic(@"
+Imports System.Threading.Tasks
+Imports System.Xml.XPath
+
+Class TestClass
+    Private Function TestMethod() As Task
+        Await Task.Run(Function() 
+        Dim xml = """"
+        Dim doc As New XPathDocument(xml)
+
+        End Function)
+    End Function
+
+    Private Sub TestMethod2()
+        Await TestMethod()
+    End Sub
+End Class",
+                GetCA3075XPathDocumentBasicResultAt(9, 20, "Run")
+            );
+        }
+
+        [Fact]
+        public void UseXPathDocumentInDelegateShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Xml.XPath;
+
+class TestClass
+{
+    delegate void Del();
+
+    Del d = delegate () {
+        var xml = """";
+        XPathDocument doc = new XPathDocument(xml);
+    };
+}",
+                GetCA3075XPathDocumentCSharpResultAt(10, 29, "TestClass")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml.XPath
+
+Class TestClass
+    Private Delegate Sub Del()
+
+    Private d As Del = Sub() 
+    Dim xml = """"
+    Dim doc As New XPathDocument(xml)
+
+End Sub
+End Class",
+                GetCA3075XPathDocumentBasicResultAt(9, 16, "TestClass")
+            );
+        }
+
+        [Fact]
+        public void UseXPathDocumentWithXmlReaderShouldNotGenerateDiagnostic()
```

Comment:
I had to convince myself that this is ok, because `XmlReader` has a `Settings` property, which in turn is an `XmlReaderSettings` object, which has a `DtdProcessing` property, whose default is `Prohibit`. That might be worth a comment here.